mobile/src/main/java/org/openhab/habdroid/util/NotificationUtils.kt

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.habdroid.util
+
+import android.annotation.TargetApi
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.os.Build
+import android.service.notification.StatusBarNotification
+import androidx.core.app.NotificationCompat
+import androidx.core.content.ContextCompat
+import org.openhab.habdroid.R
+import org.openhab.habdroid.background.NotificationUpdateObserver
+import org.openhab.habdroid.core.connection.ConnectionFactory
+import org.openhab.habdroid.model.IconResource
+import org.openhab.habdroid.ui.MainActivity
+
+object NotificationUtils {
```

Comment:
I really don't like having utility classes in Kotlin. I'll think about a better approach, probably by using a base class or composition.

Reply:
@mueller-ma This is my take on organizing the shared code, please have a look and let me know about your thoughts: https://github.com/maniac103/openhab.android/commit/96fa360b9e36fe848691bd72ed78475fdd53311b